mypy/types.py

```diff
@@ -2416,13 +2416,14 @@ def copy_modified(
             items = self.items
         return TupleType(items, fallback, self.line, self.column)
 
-    def slice(self, begin: int | None, end: int | None, stride: int | None) -> TupleType:
+    def slice(
+        self, begin: int | None, end: int | None, stride: int | None, *, fallback: Instance | None
```

Comment:
fallback should always be `named_type("builtins.tuple")` but I'm not sure how to make one without access to a Checker instance